### PR TITLE
max depth arg

### PR DESCRIPTION
**Pull request by willmcgugan**
Adds a max_depth arg to pretty printing.

### CODE REVIEW

File: tests/test_pretty.py [hunk 1]
```diff
@@ -131,6 +131,75 @@ def test_recursive():
     assert result == expected
 
 
+def test_max_depth():
+    d = d = {}
```
Comment (darrenburns) [inline tests/test_pretty.py:135]:
```suggestion
    d = {}
```

File: tests/test_pretty.py [hunk 2]
```diff
@@ -131,6 +131,75 @@ def test_recursive():
...
+    d["foo"] = {"fob": {"a": [1, 2, 3], "b": {"z": "x", "y": ["a", "b", "c"]}}}
+
+    assert pretty_repr(d, max_depth=0) == "..."
```
Comment (darrenburns) [inline tests/test_pretty.py:138]: I'd probably check the case `max_depth=None` here.

File: rich/pretty.py [hunk 3]
```diff
@@ -774,6 +805,7 @@ def pprint(
         max_length (int, optional): Maximum length of containers before abbreviating, or None for no abbreviation.
             Defaults to None.
         max_string (int, optional): Maximum length of strings before truncating, or None to disable. Defaults to None.
+        max_depth (int, optional): Maximum depth for nested data structures, or None for no maximum. Defaults to None.
```
Comment (darrenburns) [inline rich/pretty.py:808]: This sounds clearer to me
```suggestion
        max_depth (int, optional): Maximum depth for nested data structures, or None for unlimited depth. Defaults to None.
```

**Review by darrenburns — CHANGES_REQUESTED** [review]

File: tests/test_pretty.py (hunk 2 above)
Comment (willmcgugan) [reply tests/test_pretty.py:138]: Good point

**Review by darrenburns — APPROVED** [review]

**Pull request merged**